sw/device/lib/runtime/ibex.h

```diff
@@ -20,22 +20,29 @@
 extern const size_t kIbexClockFreqHz;
 
 /**
- * Spins for roughly the number of |cycles| given. For best results, |cycles|
- * should be a multiple of eight.
+ * Read the cycle counter.
  *
- * This function should not be used for time-keeping.
+ * The value of the counter is stored across two 32-bit registers: |mcycle| and
+ * |mcycleh|. This function is guaranteed to return a valid 64-bit cycle
+ * counter value, even if |mcycle| overflows before reading |mcycleh|.
  *
- * @param cycles the number of cycles to burn.
+ * Adapted from: The RISC-V Instruction Set Manual, Volume I: Unprivileged ISA
+ * V20191213, pp. 61.
  */
-inline void ibex_busy_loop(size_t cycles) {
-  size_t out;  // Used to create an inout parameter below.
+inline uint64_t read_cycle_counter() {
```

Comment:
I think this should be `ibex_cycle_counter_read`. I know it's not ibex-specific, but `ibex.h` is where this kind of stuff is going for now.

Reply:
`ibex_mcycle_read`, @imphil thoughts?

Reply:
Sure, sounds good.

Reply:
Done.